### ISSUE TITLE
[no-unnecessary-condition] optional chaining clarification

### ISSUE THREAD

**Issue by TomPridham** [typescript-eslint/typescript-eslint#1977]
i just want some clarification on this bullet point in https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md .
> Base values of optional chain expressions

 does that mean it will only check the first value in a chain?
e.g. `foo?.bar`
or will it continue to check down the chain but stop at qux?
e.g. `foo?.bar?.baz?.qux`

**Comment by bradzacher** [comment]
I just looked into this.
Interestingly, the rule incorrectly false negatives on 

```ts
type T = { a: { b: { c: string } } }
declare const x: T | null;
x?.a?.b?.c;
```

why? Because it inspects the type of the expression being optional chained.
The type of `x?.a` is `T['a'] | undefined`.
So as far as the current logic is concerned, that's a valid optional chain target.

It should be possible to work around this though, by getting the concrete type of the property (something along the lines of):
```
function checkOptionalChain(node: TSESTree.OptionalMemberExpression) {
  const memberExpr = esTreeNodeToTSNodeMap.get(node);
  const prop = memberExpr.name;
  const propSymbol = checker.getSymbolAtLocation(prop);
  const concretePropType = checker.getTypeOfSymbolAtLocation(propSymbol, memberExpr);
  
  if (!typeIsNullish(concretePropType)) {
    reportUnnecessaryOptionalChain();
  }
}
```

**Comment by moulikcipherX** [comment]
Hy @bradzacher I want to confirm about the issue that we have to make these conditions valid optional chaining.
```js 
type T = { a: { b: { c: string } } }
declare const x: T | null;
x?.a?.b?.c;
```

**Comment by bradzacher** [comment]
correct.

```ts
declare const x: { a: { b: { c: string } } } | null;
    x?.a?.b?.c;
//   ^^ valid
//      ^^ ^^ both error

declare const y: { a: { b?: { c: string } } } | null;
    y?.a?.b?.c;
//   ^^    ^^ valid
//      ^^ error
```
